prusti-interface/src/specs/typed.rs

```diff
@@ -5,24 +5,29 @@ use prusti_specs::specifications::common;
 use rustc_hash::FxHashMap;
 use prusti_rustc_interface::hir::def_id::{DefId, LocalDefId};
 use prusti_rustc_interface::span::Span;
+use prusti_rustc_interface::middle::mir;
+use prusti_rustc_interface::macros::{TyEncodable, TyDecodable};
 use std::{
     collections::HashMap,
     fmt::{Debug, Display, Formatter},
 };
+use std::rc::Rc;
 
 /// A map of specifications keyed by crate-local DefIds.
 #[derive(Default, Debug, Clone)]
-pub struct DefSpecificationMap {
+pub struct DefSpecificationMap<'tcx> {
     pub proc_specs: HashMap<DefId, SpecGraph<ProcedureSpecification>>,
     pub loop_specs: HashMap<DefId, LoopSpecification>,
     pub type_specs: HashMap<DefId, TypeSpecification>,
     pub prusti_assertions: HashMap<DefId, PrustiAssertion>,
     pub prusti_assumptions: HashMap<DefId, PrustiAssumption>,
     pub ghost_begin: HashMap<DefId, GhostBegin>,
     pub ghost_end: HashMap<DefId, GhostEnd>,
```

Comment:
These four fields (and `loop_specs`) should not need to be exported, so I was surprised to see that `PrustiAssertion` etc all got the `TyEncodable` derives too. (I see that such a subset is what you called `Lite`.)

Reply:
Right, thanks for this, fixed!